Migration.Toolkit.Core/Mappers/CmsClassMapper.cs

```diff
@@ -288,130 +162,79 @@ private void MapClassDocumentTypeFields(CmsClass source, DataClassInfo target)
-                        break;
-                    }
-
-                    case Kx13FieldDataType.Xml:
-                    {
-                        field.DataType = FieldDataType.Xml;
                         break;
                     }
-
-                    case Kx13FieldDataType.Decimal:
+                    case TcaDirective.ClearMacroTable:
                     {
-                        field.DataType = FieldDataType.Decimal;
-                        break;
-                    }
-
-                    case Kx13FieldDataType.TimeSpan:
-                    {
-                        field.DataType = FieldDataType.TimeSpan;
+                        field.SettingsMacroTable.Clear();
                         break;
                     }
-
-                    case Kx13FieldDataType.Date:
+                    case TcaDirective.ConvertToAsset:
                     {
-                        field.DataType = FieldDataType.Date;
+                        // field.DataType = FieldDataType.Assets;
+                        field.Settings["MaximumAssets"] = "99"; // setting is missing in source instance, target instance requires it
                         break;
                     }
-
-                    case Kx13FieldDataType.DocRelationships:
+                    case TcaDirective.ConvertToPages:
                     {
-                        field.Settings.Clear();
-                        field.SettingsMacroTable.Clear();
-                        field.Settings[CLASS_FIELD_CONTROL_NAME] = FormComponents.AdminPageSelectorComponent;
-                        field.Settings["MaximumPages"] = "99";
-                        field.Settings["RootPath"] = "/";
-                        field.DataType = FieldDataType.Pages;
-                        field.Size = 0;
-                        formInfo.UpdateFormField(columnName, field);
+                        field.Settings["MaximumPages"] = "99"; // setting is missing in source instance, target instance requires it
+                        field.Settings["RootPath"] = "/"; // TODO tk: 2022-08-31 describe why?
+                        field.Size = 0; // TODO tk: 2022-08-31 describe why?
+                        // field.DataType = FieldDataType.Pages;
                         break;
                     }
                 }
             }
         }
 
-        target.ClassXmlSchema = classStructureInfo.GetXmlSchema();
-        target.ClassFormDefinition = formInfo.GetXmlDefinition();
+        if (dataTypeModels.TryGetValue(columnType, out var typeMapping))
+        {
+            field.DataType = typeMapping.TargetDataType;
+            if (controlName != null &&
+                typeMapping is { FormComponents: { } } &&
+                (typeMapping.FormComponents.TryGetValue(controlName, out var targetControlMapping) ||
+                 typeMapping.FormComponents.TryGetValue("#any-not-matched-by-control-mapping-list#", out targetControlMapping)))
```

Comment:
Please, use CatchAnyNonMatching constant from SfcDirective instead of this hard-coded value.